applications/Unity.GrantManager/src/Unity.GrantManager.Application/GrantApplications/GrantApplicationAppService.cs

```diff
@@ -161,25 +163,85 @@ public async Task AddAssignee(Guid[] applicationIds, string AssigneeKeycloakId,
                 try
                 {
                     var application = await _applicationRepository.GetAsync(applicationId);
-                    if (application != null)
+                    var assignees = getAssignees(applicationId);
+                    if (application != null && (assignees == null || assignees.FindIndex(a => a.OidcSub == AssigneeKeycloakId) == -1) )
                     {
                         await _userAssignmentRepository.InsertAsync(
                             new ApplicationUserAssignment
                             {
                                 OidcSub = AssigneeKeycloakId,
-                                //ApplicationFormId = appForm1.Id,
                                 ApplicationId = application.Id,
                                 AssigneeDisplayName = AssigneeDisplayName,
                                 AssignmentTime = DateTime.Now
                             }
-                            );
+                        );
                     }
                 }
                 catch (Exception ex)
                 {
                     Debug.WriteLine(ex.ToString());
                 }
+            }
+        }
 
+        public async Task RemoveAssignee(Guid[] applicationIds, string AssigneeKeycloakId)
+        {
+            foreach (Guid applicationId in applicationIds)
+            {
+                try
+                {
+                    var application = await _applicationRepository.GetAsync(applicationId);
+                    IQueryable<ApplicationUserAssignment> queryableAssignment = _userAssignmentRepository.GetQueryableAsync().Result;
+                    var assignments = queryableAssignment.Where(a => a.ApplicationId.Equals(applicationId)).Where(b => b.OidcSub.Equals(AssigneeKeycloakId)).ToList();
+                    // Only remove the assignee if they were already assigned
+                    if (application != null && assignments != null)
+                    {
+                        await _userAssignmentRepository.DeleteAsync(assignments.FirstOrDefault());
+                    }
+                }
+                catch (Exception ex)
```

Comment:
Can remove the Exception catch block here, ABP has ways of handling the exceptions. 
Throw exception explicitly from here if needed and are created and inherited from the ABP exceptions
Could throw a EntityNotFoundException or AbpValidationException based on Validation logic here